File: rss-generator.js
const axios = require('axios');
const { create } = require('xmlbuilder2');
const fs = require('fs');
const config = require('./config.json');

const omdbApiKey = '9b8d2c00'; // استبدل بمفتاح API خاص بك من OMDb

// جلب معرف IMDb بناءً على اسم المسلسل
async function getImdbId(title) {
  try {
    const response = await axios.get(`https://www.omdbapi.com/?t=${encodeURIComponent(title)}&apikey=${omdbApiKey}`);
    if (response.data.Response === 'True' && response.data.Type === 'series') {
      return response.data.imdbID.replace('tt', '');
    }
    return null;
  } catch (error) {
    console.error('خطأ في جلب معرف IMDb:', error.message);
    return null;
  }
}

// جلب التورنتات من EZTV
async function fetchTorrents(imdbId) {
  try {
    const response = await axios.get(`https://eztvx.to/api/get-torrents?imdb_id=${imdbId}&limit=30&page=1`);
    return response.data.torrents || [];
  } catch (error) {
    console.error('خطأ في جلب التورنتات:', error.message);
    return [];
  }
}

// إنشاء ملف RSS
function generateRSS(torrents) {
  const root = create({ version: '1.0', encoding: 'UTF-8' })
    .ele('rss', { version: '2.0' })
      .ele('channel')
        .ele('title').txt('تورنتات EZTV للمسلسل').up()
        .ele('link').txt('https://medram-ram.github.io').up()
        .ele('description').txt('خلاصة RSS لتورنتات المسلسل المحدد').up()
        .ele('lastBuildDate').txt(new Date().toUTCString()).up();

  torrents.forEach(torrent => {
    root.ele('item')
      .ele('title').txt(torrent.title).up()
      .ele('link').txt(torrent.magnet_url).up()
      .ele('description').txt(`الحجم: ${(torrent.size_bytes / 1048576).toFixed(2)} MB`).up()
      .ele('pubDate').txt(new Date(torrent.date_released_unix * 1000).toUTCString()).up()
    .up();
  });

  const xml = root.end({ prettyPrint: true });
  fs.writeFileSync('rss.xml', xml);
}

// الدالة الرئيسية
async function main() {
  let imdbId = config.imdbId;
  if (!imdbId && config.title) {
    imdbId = await getImdbId(config.title);
  }
  if (imdbId) {
    const torrents = await fetchTorrents(imdbId);
    if (torrents.length > 0) {
      generateRSS(torrents);
      console.log('تم إنشاء ملف rss.xml بنجاح');
    } else {
      console.log('لم يتم العثور على تورنتات');
    }
  } else {
    console.error('لم يتم العثور على معرف IMDb صالح');
  }
}

main();

File: rss_generator.py
import requests
import json
import xml.etree.ElementTree as ET
from datetime import datetime

# مفتاح API من OMDb (استبدله بمفتاحك الخاص)
OMDB_API_KEY = '9b8d2c00'
CONFIG_FILE = 'config.json'

# جلب معرف IMDb بناءً على اسم المسلسل
def get_imdb_id(title):
    try:
        response = requests.get(f'https://www.omdbapi.com/?t={title}&apikey={OMDB_API_KEY}')
        data = response.json()
        if data.get('Response') == 'True' and data.get('Type') == 'series':
            return data['imdbID'].replace('tt', '')
        return None
    except Exception as e:
        print(f'خطأ في جلب معرف IMDb: {e}')
        return None

# جلب التورنتات من EZTV
def fetch_torrents(imdb_id):
    try:
        response = requests.get(f'https://eztvx.to/api/get-torrents?imdb_id={imdb_id}&limit=30&page=1')
        data = response.json()
        return data.get('torrents', [])
    except Exception as e:
        print(f'خطأ في جلب التورنتات: {e}')
        return []

# إنشاء ملف RSS
def generate_rss(torrents):
    rss = ET.Element('rss', version='2.0')
    channel = ET.SubElement(rss, 'channel')
    ET.SubElement(channel, 'title').text = 'تورنتات EZTV للمسلسل'
    ET.SubElement(channel, 'link').text = 'https://medram-ram.github.io'
    ET.SubElement(channel, 'description').text = 'خلاصة RSS لتورنتات المسلسل المحدد'
    ET.SubElement(channel, 'lastBuildDate').text = datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')

    for torrent in torrents:
        item = ET.SubElement(channel, 'item')
        ET.SubElement(item, 'title').text = torrent['title']
        ET.SubElement(item, 'link').text = torrent['magnet_url']
        size_mb = round(torrent['size_bytes'] / 1048576, 2)
        ET.SubElement(item, 'description').text = f'الحجم: {size_mb} MB'
        pub_date = datetime.fromtimestamp(torrent['date_released_unix']).strftime('%a, %d %b %Y %H:%M:%S GMT')
        ET.SubElement(item, 'pubDate').text = pub_date

    tree = ET.ElementTree(rss)
    tree.write('rss.xml', encoding='utf-8', xml_declaration=True)

# الدالة الرئيسية
def main():
    with open(CONFIG_FILE, 'r') as f:
        config = json.load(f)
    imdb_id = config.get('imdbId')
    if not imdb_id and config.get('title'):
        imdb_id = get_imdb_id(config['title'])
    if imdb_id:
        torrents = fetch_torrents(imdb_id)
        if torrents:
            generate_rss(torrents)
            print('تم إنشاء ملف rss.xml بنجاح')
        else:
            print('لم يتم العثور على تورنتات')
    else:
        print('لم يتم العثور على معرف IMDb صالح')

if __name__ == '__main__':
    main()